portal/static/js/accountCreation.js

```diff
@@ -183,306 +177,249 @@ var AccountCreationObj = function (roles, dependencies) {
-            self.__handleButton();
-            return false;
-        };
-        // Submit the data
-        self.treatmentCount = treatmentRows.length;
-        self.counter = 0;
-        var errorMessage = "";
+            var treatmentRows = $("#pastTreatmentsContainer tr[data-code]");
+            if (treatmentRows.length === 0) {
+                self.__handleDisplay();
+                return false;
+            }
+            if (isNaN(self.userId)) {
+                self.__handleError(i18next.t("Invalid user id: %d").replace("%d", self.userId));
+                self.__handleButton();
+                return false;
+            }
+            // Submit the data
+            self.treatmentCount = treatmentRows.length;
+            self.counter = 0;
+            var errorMessage = "";
+            treatmentRows.each(function() {
+                var procArray = {};
+                var procID = [{ "code": $(this).attr("data-code"), "display": $(this).attr("data-display"), system: $(this).attr("data-system") }];
+                procArray["resourceType"] = "Procedure";
+                procArray["subject"] = {"reference": "Patient/" + self.userId };
+                procArray["code"] = {"coding": procID};
+                procArray["performedDateTime"] = $(this).attr("data-performeddatetime");
+                (function(self) {
+                    setTimeout(function(){ tnthAjax.postProc(self.userId,procArray,false, function(data) {
+                        if (data.error) {
+                            errorMessage = data.error;
+                        }
+                        self.counter++;
+                    });}, 100);
+                })(self);
+            });
 
-        treatmentRows.each(function() {
-            var procArray = {};
-            var procID = [{ "code": $(this).attr("data-code"),
-                            "display": $(this).attr("data-display"),
-                            system: $(this).attr("data-system") }];
-            procArray["resourceType"] = "Procedure";
-            procArray["subject"] = {"reference": "Patient/" + self.userId };
-            procArray["code"] = {"coding": procID};
-            procArray["performedDateTime"] = $(this).attr("data-performeddatetime");
-            (function(self) {
-                setTimeout(function(){ tnthAjax.postProc(self.userId,procArray,false, function(data) {
-                    if (data.error) {
-                        errorMessage = data.error;
-                    };
-                    self.counter++;
-                });}, 100);
-            })(self);
-        });
+            self.treatmentIntervalVar = setInterval(function() {
+                if (self.counter === self.treatmentCount) {
+                    if (hasValue(errorMessage)) {
+                        self.__handleError(errorMessage + i18next.t(" Account created.  Redirecting to profile..."));
```

Comment:
Could you move the leading space outside the `t()` call? Looks like an extra space after the `.` too